lighthouse-core/runner.js

```diff
@@ -90,31 +89,31 @@ class Runner {
         return artifacts;
       });
 
-      // Ignoring these two flags for coverage as this functionality is not exposed by the module.
-      /* istanbul ignore next */
-      if (opts.flags.saveArtifacts) {
-        run = run.then(artifacts => {
-          opts.flags.saveArtifacts && assetSaver.saveArtifacts(artifacts);
-          return artifacts;
-        });
-      }
-      if (opts.flags.saveAssets) {
+      // Run each audit sequentially, the auditResults array has all our fine work
+      const auditResults = [];
+      for (const audit of config.audits) {
         run = run.then(artifacts => {
-          return assetSaver.saveAssets(opts, artifacts)
+          return Runner._runAudit(audit, artifacts)
+            .then(ret => auditResults.push(ret))
             .then(_ => artifacts);
         });
       }
-
-      // Run each audit sequentially, the auditResults array has all our fine work
-      const auditResults = [];
-      run = run.then(artifacts => config.audits.reduce((chain, audit) => {
-        return chain.then(_ => {
-          return Runner._runAudit(audit, artifacts);
-        }).then(ret => auditResults.push(ret));
-      }, Promise.resolve()).then(_ => auditResults));
+      run = run.then(artifacts => {
+        return {artifacts, auditResults};
+      });
     } else if (config.auditResults) {
       // If there are existing audit results, surface those here.
-      run = run.then(_ => config.auditResults);
+      // Instantiate and return artifacts for consistency.
+      // Although we can move this upwards to reduce code repetition, this process might be quite
```

Comment:
the `// Although...` lines probably aren't necessary here (though good for the code reviewer :). I do think you're right we want to have the same object shape for `artifacts` regardless of path through here